Mathlib/Combinatorics/SetFamily/CauchyDavenport.lean

```diff
@@ -0,0 +1,194 @@
+/-
+Copyright (c) 2023 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Combinatorics.Additive.ETransform
+import Mathlib.GroupTheory.Order.Min
+
+/-!
+# The Cauchy-Davenport theorem
+
+This file proves a generalisation of the Cauchy-Davenport theorem to arbitrary groups.
```

Comment:
Please expand docs (both module-level and docstrings of individual theorems) so that they are useful for a reader who doesn't know what is the Cauchy-Davenport theorem.